hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/AnyDiscriminatorSqmPath.java

```diff
@@ -67,4 +66,15 @@ public SqmTreatedPath treatAs(Class treatJavaType) throws PathException {
 	public SqmTreatedPath treatAs(EntityDomainType treatTarget) throws PathException {
 		throw new UnsupportedMappingException( "Cannot apply TREAT operator to discriminator path" );
 	}
+
+	@Override
+	public SqmPath<T> getLhs() {
+		return (SqmPath<T>) super.getLhs().getLhs();
+	}
```

Comment:
This feels a bit wrong to me, as the lhs of a `type(root.anyAttribute)` should be the path for `root.anyAttribute`, but this seems to return `root` instead.

Reply:
I have to dig into this